lib/internal/modules/esm/translators.js

```diff
@@ -128,14 +128,10 @@ translators.set('builtin', async function builtinStrategy(url) {
   if (!module) {
     throw new ERR_UNKNOWN_BUILTIN_MODULE(id);
   }
-  return createDynamicModule(
-    [], [...module.exportKeys, 'default'], url, (reflect) => {
-      debug(`Loading BuiltinModule ${url}`);
-      module.reflect = reflect;
-      for (const key of module.exportKeys)
-        reflect.exports[key].set(module.exports[key]);
-      reflect.exports.default.set(module.exports);
-    });
+  debug(`Loading BuiltinModule ${url}`);
+  const facade = module.getESMFacade();
+  module.syncExports();
```

Comment:
Can you clarify why we need to call this for each builtin load? What use cases does that enable or is it necessary for the approach to work?

If it is necessary, perhaps we can just run the sync on the module just loaded?

Reply:
this does only sync the module being loaded.

the second line actually looks like a bug though since it would sync on first `import` even if it was already loaded, which isn't desirable

Reply:
I think this is the module just loaded: `syncExports` (one) vs. `syncESMExports` (all).

Reply:
Ah I misread, thanks.